packages/beacon-node/src/network/peers/peerManager.ts

```diff
@@ -177,6 +178,10 @@ export class PeerManager {
       this.nodeId,
       Math.max(this.config.CUSTODY_REQUIREMENT, this.config.NODE_CUSTODY_REQUIREMENT, this.config.SAMPLES_PER_SLOT)
     );
+    this.samplingGroups = getCustodyGroups(
```

Comment:
Should we use the `custodyConfig` value instead of recalculating?

Reply:
yes, we should get from a global custodyConfig every time, instead of only at the constructor
and it's likely we need to work with `this.samplingGroups`, no need `this.samplingSubnets`
this is the same concern you have here so I'd work on that separately https://github.com/ChainSafe/lodestar/pull/7607#pullrequestreview-2701240021
will add a `TODO(das)` for it